packages/x-date-pickers/src/internals/constants/dimensions.ts

```diff
@@ -2,6 +2,6 @@ export const DAY_SIZE = 36;
 export const DAY_MARGIN = 2;
 export const DIALOG_WIDTH = 320;
 export const MAX_CALENDAR_HEIGHT = 280;
-export const VIEW_HEIGHT = 334;
+export const VIEW_HEIGHT = 336;
```

Comment:
This and the changes on `PickersCalendarHeader` serve in aligning the height definitions to match between DateCalendar and DateRangeCalendar.
It was evident when trying to make `DateRangeCalendar` with 1 and 2 calendars work nicely in `DesktopDateTimeRangePicker`. 🙈

Reply:
Opened up https://github.com/mui/mui-x/pull/11337 to deliver it separately and clearly split the Argos diff.